hal/reset_reason_api.h

```diff
@@ -0,0 +1,65 @@
+/** \addtogroup hal */
+/** @{*/
+/* mbed Microcontroller Library
+ * Copyright (c) 2017 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef MBED_RESET_REASON_API_H
+#define MBED_RESET_REASON_API_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef enum
+{
+  RESET_REASON_POWER_ON  = (1 << 0),
+  RESET_REASON_PIN_RESET = (1 << 1),
+  RESET_REASON_BROWN_OUT = (1 << 2),
+  RESET_REASON_SOFTWARE  = (1 << 3),
+  RESET_REASON_WATCHDOG  = (1 << 4),
+  RESET_REASON_LOCKUP    = (1 << 5),
+  RESET_REASON_UNKNOWN   = (1 << 6)
+} reset_reason_t;
```

Comment:
Could you document the enumeration values ? The left shifting of each value suggest that the reset value is a bitfield.

Reply:
The left shifting was left from when I planned to OR the reasons together in case multiple were set in the registers. I don't think that's necessary now.